docs/howto/retry.rst

```diff
@@ -34,11 +34,24 @@ Simple strategies
 Advanced strategies
 ^^^^^^^^^^^^^^^^^^^
 
-You can get a more precise strategy using a :py:class:`RetryStrategy` instance::
+Advanced strategies let you:
+
+- define a maximum number of retries (if you don't, the job will be retried
+  indefinitely until it passes)
+- define the retry delay, with constant, linear and exponential backoff options (if
+  you don't, jobs will retry immediately)
+- define the exception types you want to retry on (if you don't, retry will occur on
+  any type of exceptions)
+
+Define your precise strategy using a :py:class:`RetryStrategy` instance::
 
     from procrastinate import RetryStrategy
 
-    @app.task(retry=procrastinate.RetryStrategy(max_attempts=10, wait=5))
+    @app.task(retry=procrastinate.RetryStrategy(
+        max_attempts=10,
+        wait=5,
+        retry_exceptions=[ConnectionError, IOError]
```

Comment:
I'd use a `tuple` here instead of a `list`. To advertise better practices to our users.

Reply:
In what way is this a better practice to use a tuple here ? This is an array of homogeneous elements with arbitrary length, which fits better as a list IMHO, contrary to a tuple which works best as a fixed-length iterable where each element's type and signification is described through their position, no ?

Reply:
To be more precise, my rule of thumb:
- If you have very specific performance constraints, you may explore the collections zoo (deque, ...)
- If it makes no sense to have duplicates and order doesn't matter, it's a set
- If any 2 elements could be swapped and it would still make sense, it's a list
- if it can be used as basis for a namedtuple, it's a tuple (and you should consider using a namedtuple)
- Else... I'm interested to know what iterable you've found that fail all of the above tests.

Reply:
I did not go that far :-)

I just think a fixed-length, immutable, data structure makes more sense in such cases. In the same way `isinstance` receives a tuple as the second argument. Or when you write `elem in ("foo", "bar")`.

But I don't mind to be too pedantic. Using a list is fine too.

Reply:
Ha sorry for being pedantic too :D

I think the fact `ìsinstance()` takes a tuple and not an iterable is an error from Python's youth.

Also, I think you were totally right questioning my choice of using a list: from my own rationale just above, a set would actually be much more appropriate (order doesn't matter and duplicates make no sense) 😆

And finally, regarding `elem in ("foo", "bar")`, I was curious if the most "appropriate" structure (here, a set) was also the most performant and it kinda looks like it is (with huge question marks regarding the validity of my benchmark):
```
In [1]: %timeit "baz" in {"foo", "bar"}
29.6 ns ± 0.79 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [2]: %timeit "baz" in ("foo", "bar")
57 ns ± 0.656 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [3]: %timeit "baz" in ["foo", "bar"]
58 ns ± 1.9 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```
(this holds true if we're looking for the 2nd element but the 3 are as long if we're looking for the 1st element)
(Just to be clear, I'm not advocating for using one or another way just to earn a few nanoseconds as long as it's not on the critical path)